### PR TITLE
bugfix: pyTenable crashes if srcInterface or dstInterface not null

### PR DESCRIPTION
**Pull request by saizinbud**
# Description

Addressed a bug causing a crash in when events->srcInterface or events->dstInterface was not null upon query completion. The issue was traced back to incorrect usage of the marshmallow package.

Note: In Tenable.OT version 3.17, a bug existed where events->srcInterface and dstInterface consistently returned null. Due to this behavior, the current bug did not manifest itself during that version

Additionally, added the retrieval of required nodes from dstInterface, ensuring that all necessary nodes are now properly obtained.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Run local tests and tested against a Tenable.OT machine

**Test Configuration**:
* Python Version(s) Tested: 3.10.12

### CODE REVIEW

**Review by elad-indegy — APPROVED** [review]

File: tenable/ot/schema/assets.py [hunk 1]
```diff
@@ -144,16 +145,27 @@ class Assets(NodesList):
     nodes: List[Asset]
 
 
+@dataclass
```
Comment (AnavShapiraTenable) [inline tenable/ot/schema/assets.py:148]: I think that you can move it to tenable/ot/schema/base.py (the IP calss and the IPList)

Comment (saizinbud) [reply tenable/ot/schema/assets.py]: Done

File: tenable/ot/schema/base.py [hunk 2]
```diff
@@ -32,3 +33,13 @@ class AssetInfo:
 @dataclass
 class AssetInfoList(NodesList):
     nodes: List[AssetInfo]
+
+
+@dataclass
+class IP:
```
Comment (AnavShapiraTenable) [inline tenable/ot/schema/base.py:39]: Use this class also in tenable/ot/schema/events.py line:124 (dst_ip) and on line:139 (src_ip)

Comment (saizinbud) [reply tenable/ot/schema/base.py:39]: Done

**Review by elad-indegy — APPROVED** [review]

**Comment by aseemsavio** [comment]
@saizinbud The changes look good to me. Could you also add tests?

**Review by aseemsavio — APPROVED** [review]

**Review by varap11 — APPROVED** [review]

**Pull request merged**